bnxt_re/lib: Enable Broadcom's 57500 RoCE adapter

dsharma283 opened the pull request:
This is to add Broadcom's 57500 series of adapters support
to RoCE from libbnxt_re. Listing below the significant changes
done as part of the patch.

 - Added the pci-id of the basic gen-p5 chip.
 - Adjust psn search memory allocation to suite new search
   psn structure.
 - Added chip context structure to select the appropriate
   execution flow in data-path and control path.
 - bumped up the ABI version to 2.
 - Fill psn search area as per new or older chip execution
   flow.
 - removed duplicate declaration of BNXT_RE_ABI_VERSION macro

Signed-off-by: Devesh Sharma <devesh.sharma@broadcom.com>

jgunthorpe commented:
Why are you bumping the ABI version? I don't want to see this.

dsharma283 commented:
> 
> 
> Why are you bumping the ABI version? I don't want to see this.

Oh, in your reply to my kernel patches did you mean that because the new members are always being added at the end of the structure, there is no need to bump ABI version? Just check for an invalid-value-tag in the new filed(s) and assume to be working with older ABI because the value was invalid?

I might have missed to understand your complete meaning.

jgunthorpe commented:
yes something like that. Usually you should add a 'comp_mask' as part of your new fields then !0 comp_mask means the new fields are supported

dsharma283 commented:
Could this also be pulled, or need a rev?

jgunthorpe commented:
This is not properly formed, the kernel headers have to be updated using the kernel-headers/update 
script.

dsharma283 commented:
> 
> 
> This is not properly formed, the kernel headers have to be updated using the kernel-headers/update
> script.

Could you give a pointer on "how-to" I would repost asap.

jgunthorpe commented:
https://www.spinics.net/lists/linux-rdma/msg72030.html

dsharma283 commented:
> 
> 
> https://www.spinics.net/lists/linux-rdma/msg72030.html

Jason,

I guess I need additional comment form you on this, I tried to use "update" script but it is not generating the correct patch. Noteworthy part is, the kernel patches are already accepted and in the tree, I used following both ways but no luck.

Method 1:

- rebased kernel-git-tree to the commit id where abi changes were there
- run update script ./kernel-headers/update  --not-final /data2/upstream/korg-rdma HEAD
The result was incorrect patch

Method 2:
- sync the kernel-git-tree with the upstream for-next branch
- run update script with the commit-id
./kernel-headers/update  /data2/upstream/korg-rdma 95b86d1c91ad3b19f882d9e70aa37c8e99e8dc17

Result was incorrect patch. I am not sure what's missing?

dsharma283 commented:
> 
> 
> > https://www.spinics.net/lists/linux-rdma/msg72030.html
> 
> Jason,
> 
> I guess I need additional comment form you on this, I tried to use "update" script but it is not generating the correct patch. Noteworthy part is, the kernel patches are already accepted and in the tree, I used following both ways but no luck.
> 
> Method 1:
> 
>     * rebased kernel-git-tree to the commit id where abi changes were there
> 
>     * run update script ./kernel-headers/update  --not-final /data2/upstream/korg-rdma HEAD
>       The result was incorrect patch
> 
> 
> Method 2:
> 
>     * sync the kernel-git-tree with the upstream for-next branch
> 
>     * run update script with the commit-id
>       ./kernel-headers/update  /data2/upstream/korg-rdma 95b86d1c91ad3b19f882d9e70aa37c8e99e8dc17
> 
> 
> Result was incorrect patch. I am not sure what's missing?

Looks like the correct patch is not being generated because the kernel-header changes has already been merged into rdma-core code base. I have rebase this pull request and dropped the kernel-abi related changes from the patch. Please pull this patch and merge.

Let me know if anything else is needed.

rleon commented:
What is the status?

dsharma283 commented:
Hi Leon,

There was an ask from Jason to update the kernel abi changes using update script. However, in the latest rdma core those.changes has already been merged
Therefore i have rebased this patch and dropped the kernel abi related change. I guess if there are no furher comments this can be merged

rleon commented:
so why do I still see changes in -abi.h file?
https://github.com/linux-rdma/rdma-core/pull/453/commits/a9f35f4f8e24b294685f30bdc90b14a23c6d403d

rleon commented:
Ohh, I see, it is providers/*abi.h
You guys have special place in my heart for choosing names.
